client/network/src/lib.rs

```diff
@@ -454,3 +457,106 @@ async fn wait_to_announce<Block: BlockT>(
 		);
 	}
 }
+
+#[derive(Clone)]
+pub struct WaitForParachainTargetBlock<Block> {
```

Comment:
I don't get why this is a `struct` and not just a function? The functionality should also be moved into its own file.